autogpt_platform/backend/test/integration/test_virus_scanner.py

```diff
@@ -0,0 +1,223 @@
+import asyncio
+import time
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+
+from backend.server.v2.store.exceptions import VirusDetectedError, VirusScanError
+from backend.util.virus_scanner import (
+    VirusScannerService,
+    VirusScannerSettings,
+    VirusScanResult,
+    get_virus_scanner,
+    scan_content_safe,
+)
+
+
+class TestVirusScannerService:
+    @pytest.fixture
+    def scanner_settings(self):
+        return VirusScannerSettings(
+            clamav_service_host="localhost",
+            clamav_service_port=3310,
+            clamav_service_enabled=True,
+            max_scan_size=10 * 1024 * 1024,  # 10MB for testing
+        )
+
+    @pytest.fixture
+    def scanner(self, scanner_settings):
+        return VirusScannerService(scanner_settings)
+
+    @pytest.fixture
+    def disabled_scanner(self):
+        settings = VirusScannerSettings(clamav_service_enabled=False)
+        return VirusScannerService(settings)
+
+    def test_scanner_initialization(self, scanner_settings):
+        scanner = VirusScannerService(scanner_settings)
+        assert scanner.settings.clamav_service_host == "localhost"
+        assert scanner.settings.clamav_service_port == 3310
+        assert scanner.settings.clamav_service_enabled is True
+
+    @pytest.mark.asyncio
+    async def test_scan_disabled_returns_clean(self, disabled_scanner):
+        content = b"test file content"
+        result = await disabled_scanner.scan_file(content, filename="test.txt")
+
+        assert result.is_clean is True
+        assert result.threat_name is None
+        assert result.file_size == len(content)
+        assert result.scan_time_ms == 0
+
+    @pytest.mark.asyncio
+    async def test_scan_file_too_large(self, scanner):
+        # Create content larger than max_scan_size
+        large_content = b"x" * (scanner.settings.max_scan_size + 1)
+
+        # Large files are allowed but marked as clean with a warning
+        result = await scanner.scan_file(large_content, filename="large_file.txt")
+        assert result.is_clean is True
```

Comment:
Uh

Reply:
Practicing what they preach 

https://support.google.com/drive/thread/225474267/virus-scanning-files-in-google-drive-larger-than-100mb?hl=en